simple-local-avatars.php

```diff
@@ -1,571 +1,610 @@
+		$user_id = (int) $_POST['user_id'];
+
+		// ensure the media is real is an image
+		if ( wp_attachment_is_image( $media_id ) )
+			$this->assign_new_user_avatar( $media_id, $user_id );
+
+		echo get_simple_local_avatar( $user_id );
+
+		die;
+	}
+
+	/**
+	 * remove the custom get_avatar hook for the default avatar list output on options-discussion.php
+	 */
+	public function avatar_defaults( $avatar_defaults ) {
+		remove_action( 'get_avatar', array( $this, 'get_avatar' ) );
+		return $avatar_defaults;
+	}
+
+	/**
+	 * Delete avatars based on a user_id
+	 *
+	 * @param int $user_id
+	 */
+	public function avatar_delete( $user_id ) {
+		$old_avatars = (array) get_user_meta( $user_id, 'simple_local_avatar', true );
+
+		if ( empty( $old_avatars ) )
+			return;
+
+		// if it was uploaded media, don't erase the full size or try to erase an the ID
+		if ( array_key_exists( 'media_id', $old_avatars ) )
+			unset( $old_avatars['media_id'], $old_avatars['full'] );
+
+		if ( ! empty( $old_avatars ) ) {
+			$upload_path = wp_upload_dir();
+
+			foreach ($old_avatars as $old_avatar ) {
+				// derive the path for the file based on the upload directory
+				$old_avatar_path = str_replace( $upload_path['baseurl'], $upload_path['basedir'], $old_avatar );
+				if ( file_exists( $old_avatar_path ) )
+					unlink( $old_avatar_path );
+			}
+		}
+
+		delete_user_meta( $user_id, 'simple_local_avatar' );
+		delete_user_meta( $user_id, 'simple_local_avatar_rating' );
+	}
+
+	/**
+	 * Creates a unique, meaningful file name for uploaded avatars.
+	 *
+	 * @param string $dir Path for file
+	 * @param string $name Filename
+	 * @param string $ext File extension (e.g. ".jpg")
+	 * @return string Final filename
+	 */
+	public function unique_filename_callback( $dir, $name, $ext ) {
+		$user = get_user_by( 'id', (int) $this->user_id_being_edited );
+		$name = $base_name = sanitize_file_name( $user->display_name . '_avatar_' . time() );
+
+		// ensure no conflicts with existing file names
+		$number = 1;
+		while ( file_exists( $dir . "/$name$ext" ) ) {
+			$name = $base_name . '_' . $number;
+			$number++;
+		}
+
+		return $name . $ext;
+	}
+
+	/**
+	 * Adds errors based on avatar upload problems.
+	 *
+	 * @param WP_Error $errors Error messages for user profile screen.
+	 */
+	public function user_profile_update_errors( WP_Error $errors ) {
+		$errors->add( 'avatar_error', $this->avatar_upload_error );
+	}
+
+	/**
+	 * Registers the simple_local_avatar field in the REST API.
+	 */
+	public function register_rest_fields() {
+		register_rest_field( 'user', 'simple_local_avatar', array(
+			'get_callback' => array( $this, 'get_avatar_rest' ),
+			'update_callback' => array( $this, 'set_avatar_rest' ),
+			'schema' => array(
+				'description' => 'The users simple local avatar',
+				'type' => 'object',
+			)
+		));
+	}
+
+	/**
+	 * Returns the simple_local_avatar meta key for the given user
+	 * @param object $profileuser User object
```

Comment:
I'm guessing this param is supposed to be `$user`? And the one below a leftover from something else that isn't needed anymore.